js/ssai_pulse.js

```diff
@@ -372,12 +388,34 @@ OO.Ads.manager(function(_, $)
      * @param {XMLDocument} xml The xml returned from loading the ad
      * @param {object} id3Object The ID3 object
      */
-    this.onResponse = function(xml, id3Object)
+    this.onResponse = function(id3Object, xml)
     {
       OO.log("SSAI Pulse: Response");
       // Call VastParser code
-      // var vastAds = OO.VastParser.parser(xml);
-      _forceMockAd(id3Object);
+      var vastAds = OO.VastParser.parser(xml);
+      var adIdVastData = _parseVastAdsObject(vastAds);
+
+      var ssaiAd =
+      {
+        clickthrough: "",
+        name: "",
+        ssai: true,
+        isLive: true
+      };
+
+      if (_.has(adIdVastData, id3Object.adId))
+      {
+        var adObject = adIdVastData[id3Object.adId];
+        this.adIdDictionary[id3Object.adId].vastData = adObject;
+        ssaiAd.data = adObject;
+        ssaiAd.clickthrough = _getLinearClickThroughUrl(adObject);
+        ssaiAd.name = _getTitle(adObject);
+      }
+
+      this.adIdDictionary[id3Object.adId] = STATE.PLAYED;
```

Comment:
PLAYED sounds like the ad has already finished playing. Should we rename this state to PLAYING? What are your thoughts?